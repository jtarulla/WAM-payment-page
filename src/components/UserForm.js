import React, { Component } from 'react';
import nextButton from '../assets/buttons/btn-next.png';
import page from '../assets/page-1.png';

export class UserForm extends Component {
	continue = e => {
		e.preventDefault();
		const { values } = this.props;

		if (values.firstName !== '' && values.lastName !== '') {
			this.props.nextStep();
		}
	};

	render() {
		const { values, handleChange } = this.props;
		return (
			<>
				<div className='page-number'>
					<img src={page} alt='page-number' />
				</div>
				<div className='container'>
					<div className='form-area'>
						<h2>Mis Datos</h2>
						<p>
							Revisa los datos y completa aquellos pendientes antes de continuar
							con el proceso.
						</p>
						<br />
						<form>
							<label>
								Nombre
								<input
									type='text'
									id='firstName'
									name='firstName'
									onChange={handleChange('firstName')}
									defaultValue={values.firstName}
									autoFocus={true}
									required
								/>
							</label>
							<label>
								Apellidos
								<input
									type='text'
									id='lastName'
									name='lastName'
									onChange={handleChange('lastName')}
									defaultValue={values.lastName}
									required
								/>
							</label>
						</form>
					</div>

					<div className='btn-continue' onClick={this.continue}>
						<img src={nextButton} alt='pay-button' />
					</div>
				</div>
			</>
		);
	}
}

export default UserForm;
