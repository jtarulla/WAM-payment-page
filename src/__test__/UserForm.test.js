import { mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import UserForm from '../components/UserForm';

describe('UserForm component', () => {
	const props = {
		nextStep: jest.fn(),
		handleChange: jest.fn(),
		values: {
			firstName: 'First',
			lastName: 'Last'
		}
	};
	it('renders without error given the required props', () => {
		const wrapper = mount(<UserForm {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should call nextStep prop when clicking [.btn-continue]', () => {
		const wrapper = mount(<UserForm {...props} />);
		const button = wrapper.find('.btn-continue');
		button.simulate('click', {
			preventDefault: () => {}
		});
		const callback = props.nextStep.mock.calls.length;
		expect(callback).toBe(1);
	});

	it('should call handleChange prop when changing first name input', () => {
		const wrapper = mount(<UserForm {...props} />);
		const input = wrapper.find('input#firstName');
		input.simulate('change', { target: { id: 'firstName', value: 'John' } });
		expect(props.handleChange).toHaveBeenCalled();
	});

	it('should call handleChange prop when changing last name input', () => {
		const wrapper = mount(<UserForm {...props} />);
		const input = wrapper.find('input#lastName');
		input.simulate('change', { target: { id: 'lastName', value: 'Doe' } });
		expect(props.handleChange).toHaveBeenCalled();
	});
});
