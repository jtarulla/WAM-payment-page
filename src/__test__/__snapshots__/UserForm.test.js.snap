// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm component renders without error given the required props 1`] = `
<UserForm
  handleChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "firstName",
        ],
        Array [
          "lastName",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  nextStep={[MockFunction]}
  values={
    Object {
      "firstName": "First",
      "lastName": "Last",
    }
  }
>
  <div
    className="page-number"
  >
    <img
      alt="page-number"
      src="page-1.png"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="form-area"
    >
      <h2>
        Mis Datos
      </h2>
      <p>
        Revisa los datos y completa aquellos pendientes antes de continuar con el proceso.
      </p>
      <br />
      <form>
        <label>
          Nombre
          <input
            autoFocus={true}
            defaultValue="First"
            id="firstName"
            name="firstName"
            required={true}
            type="text"
          />
        </label>
        <label>
          Apellidos
          <input
            defaultValue="Last"
            id="lastName"
            name="lastName"
            required={true}
            type="text"
          />
        </label>
      </form>
    </div>
    <div
      className="btn-continue"
      onClick={[Function]}
    >
      <img
        alt="next-button"
        src="btn-next.png"
      />
    </div>
  </div>
</UserForm>
`;
